pkg/pillar/cmd/ledmanager/ledmanager.go

```diff
@@ -432,22 +441,26 @@ func handleLedBlinkDelete(ctxArg interface{}, key string,
 
 // handleDisplayUpdate waits for changes and displays/blinks the based on
 // the updated counter
-func handleDisplayUpdate(countChange chan int, displayFunc DisplayFunc,
-	arg string) {
+func handleDisplayUpdate(ctx *ledManagerContext, displayFunc DisplayFunc,
+	arg string, isDisplay bool) {
 
 	var counter int
 	for {
+		changed := false
 		select {
-		case counter = <-countChange:
+		case counter = <-ctx.countChange:
 			log.Tracef("Received counter update: %d",
 				counter)
+			changed = true
 		default:
 			log.Tracef("Unchanged counter: %d", counter)
 		}
-		log.Tracef("Displaying counter %d", counter)
-		// XXX separate logic if display? No need to update every 1200ms
 		if displayFunc != nil {
-			displayFunc(arg, counter)
+			log.Tracef("Displaying counter %d", counter)
+			// Skip unchanged updates if it is a true display
+			if changed || !isDisplay {
```

Comment:
is it meant to be 
```
if changed && isDisplay 
```
?

If yes, then I guess the above `if` could be
```
if displayFunc && changed && isDisplay
```

Reply:
No. If there is a change we always display. And if there is no change we still need to call the function to have it do the next blink sequence.